action/protocol/staking/candidate_statereader.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2020 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package staking
+
+import (
+	"github.com/iotexproject/iotex-core/action/protocol"
+	"github.com/iotexproject/iotex-core/state"
+	"github.com/pkg/errors"
+)
+
+type (
+	// CandidateStateReader contains candidate center and bucket pool
+	CandidateStateReader interface {
+		// TODO: remove CandidateCenter interface, return *candCenter
+		Height() uint64
+		CandCenter() CandidateCenter
+		BucketPool() *BucketPool
```

Comment:
If we do return CandidateCenter and BucketPool, CandidateStateReader is not a real reader.